packages/astro/src/@types/astro.ts

```diff
@@ -1094,6 +1094,29 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		inlineStylesheets?: 'always' | 'auto' | 'never';
+		/**
+		 * @docs
+		 * @name build.concurrency
+		 * @type {boolean}
+		 * @default `1`
+		 * @description
+		 * The number of pages to build in parallel. In most cases, you should stick with the default value of `1`,
+		 * and batch or cache long running tasks like fetch calls or data access to improve the overall rendering time.
+		 *
+		 * Use this option only if the refactors are not possible. If the number is set too high, the page rendering
+		 * may slow down due to insufficient memory resources and because JS is single-threaded.
```

Comment:
```suggestion
		 * The number of pages to build in parallel.
		 * 
		 * **In most cases, you should not change the default value of `1`.**
		 * 
		 * Use this option only when other attempts to reduce the overall rendering time (e.g. batch or cache long running tasks like fetch calls or data access) are not possible or are insufficient.
		 * If the number is set too high, page rendering may slow down due to insufficient memory resources and because JS is single-threaded.
```
I think maybe structuring the docs like this makes it clear first, *what* this is, and then **warns you not to do this** as a simple way to speed up your project.